cluster/kubernetes/kubernetes.go

```diff
@@ -89,55 +87,42 @@ func isAddon(obj namespacedLabeled) bool {
 // --- /add ons
 
 type Applier interface {
-	Delete(logger log.Logger, def *apiObject) error
-	Apply(logger log.Logger, def *apiObject) error
+	stage(id, cmd string, obj *apiObject)
+	execute(log.Logger, cluster.SyncError)
 }
 
 // Cluster is a handle to a Kubernetes API server.
 // (Typically, this code is deployed into the same cluster.)
 type Cluster struct {
 	client     extendedClient
 	applier    Applier
-	actionc    chan func()
 	version    string // string response for the version command.
 	logger     log.Logger
 	sshKeyRing ssh.KeyRing
+
+	sync.Mutex
```

Comment:
If this isn't be to be used outside the methods of Cluster, it should be given an unexported name.

Reply:
Fixed.